jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -582,6 +671,26 @@ ecma_utf8_string_to_number (const lit_utf8_byte_t *str_p, /**< utf-8 string */
 
       e_in_lit = e_in_lit * 10 + digit_value;
 
+      /**
+       * Sanity check: If the exponent part is bigger than 10^3 we should return with Infinity
+       * because log10(Number.MAX_VALUE) ~= 308.2547.
+       * If the exponent part is less than -10^3  we should return with 0
+       * because log10(Number.MIN_VALUE) ~= -323.3062.
+       * In addition this check prevents e_in_lit overflow even
+       * if the convertible number's exponent part bigger than UINT32_MAX.
+       */
+      if (exponent_length > 3)
```

Comment:
Proposal:
```c
int32_t e_check = e;
if (e_in_lit_sign)
{
  e_check -= e_in_lit;
}
else
{
  e_check += e_in_lit;
}
 
if (e_check < NUMBER_MIN_DECIMAL_EXPONENT)
{
  return sign ? -ECMA_NUMBER_ZERO : ECMA_NUMBER_ZERO;
}
if (e_check >NUMBER_MAX_DECIMAL_EXPONENT)
{
  return ecma_number_make_infinity (sign);
}
```
Wouldn't this solve all the issues (actual exponent value, float size, int overflow)?

And I think that no additional code would be needed (helper function and its invocation additions below).

Reply:
That's not a bad idea, but E.g. the number is `1.797693134862315708145274237317e+308` which mustn't be Infinity.
In this case `e = 290` and `e_check = 598` so it gives Infinity. That's a false result, but with a little bit of refinement, it may could work.

Reply:
I don't think that the numbers above are right. `e` is first computed from the number digits only - in this case from  `1.7976...`, giving now a negative number (-30 perhaps?). And `e_check` will be `e + e_in_lit`, -30 + 308 = 278.

This reveals an issue with my proposal, though. I haven't taken floor(log10(fraction_uin64)) into account. `e_check` has to be adjusted with that.

Am I right now?

Reply:
I believe we misunderstood each other. Would you use your proposed code instead of the 'calculating the length of the exponent and return if it's longer than 3'?  Or after this part? 

So my question is where would you like to add your proposed code exactly?